DOL.WHD.Section14c.Log/Repositories/ErrorLogRepository.cs

```diff
@@ -0,0 +1,79 @@
+﻿using DOL.WHD.Section14c.Log.DataAccess;
+using DOL.WHD.Section14c.Log.Models;
+using NLog;
+using System;
+using System.Linq;
+
+namespace DOL.WHD.Section14c.Log.DataAccess.Repositories
+{
+    public class ErrorLogRepository: IErrorLogRepository
+    {
+        private readonly ApplicationLogContext _dbContext;
+        private static readonly Logger _logger = LogManager.GetCurrentClassLogger();
+        private Boolean Disposed;
+
+        public ErrorLogRepository()
+        {
+            _dbContext = new ApplicationLogContext();
+        }
+
+        /// <summary>
+        /// Get All Logs
+        /// </summary>
+        /// <returns></returns>
+        public IQueryable<APIErrorLogs> GetAllLogs()
+        {
+            return _dbContext.ErrorLogs.AsQueryable();
+        }
+
+        /// <summary>
+        /// Add New Log Message
+        /// </summary>
+        /// <param name="entity"></param>
+        /// <returns></returns>
+        public LogDetails AddLog(LogDetails entity)
+        {
+            
+            if (entity != null)
+            {
+                LogEventInfo eventInfo = new LogEventInfo();
+                eventInfo.Properties[Constants.CorrelationId] = Guid.NewGuid().ToString();
+
+                eventInfo.Properties[Constants.EIN] = string.IsNullOrEmpty(entity.EIN) ? string.Empty : entity.EIN;
+                eventInfo.LoggerName = "NLog";
+                if (string.IsNullOrEmpty(entity.Message))
+                {
+                    throw new ArgumentException("Message cannot be null or empty string", "Log Message");
+                }
```

Comment:
I'm thinking this if statement should be moved up to line 39, so we don't even create the `LogEventInfo` object if the call is invalid.  Fail as early as possible.

Also, would this `ArgmuentException` get logged?  🤔

Reply:
Yes, It will get logged.